server/app/core/usecases.py

```diff
@@ -144,61 +156,296 @@ async def save(
+                )
+            else:
+                raise ConnectorError(
+                    f"Connector returned {response.status_code} for {access_url_str}"
+                )
+
+            # Merge connector values in-place
+            self._update_distribution_in_place(
+                dataset.graph, dist_node, connector_data, existing_data
+            )
+
+        # Add dataset-level metadata
         is_shared = dataset.get_attribute(DSPACE.isShared)
         if is_shared is None:
             dataset.set_attribute(DSPACE.isShared, False)
         dataset.set_attribute(DCTERMS.issued, datetime.now(UTC).isoformat())
         dataset.set_attribute(DSPACE.isDeleted, False)
         dataset.set_attribute(DCTERMS.publisher, person.uri)
 
-        # Set the dataset as a child of the catalog
-        catalog.set_attribute(DCAT.dataset, dataset.uri)
+        # Extracting region from catalog
+        catalog_title = catalog.get_attribute(DCTERMS.title)
+        region_value = catalog_title or "Unknown Region"
+        dataset.set_attribute(DSPACE.region, region_value)
 
-        # Save the graph (including the dataset, catalog, and person)
+        # Link dataset to catalog and save
+        catalog.set_attribute(DCAT.dataset, dataset.uri)
         catalog += dataset
         catalog += person
+
         await self.repositories.catalogs.save(catalog)
 
-        # Return the dataset
-        id = dataset.get_attribute(DCTERMS.identifier)
-        final_dataset = await self.get(id, context)
+        #  Return updated dataset
+        dataset_id = dataset.get_attribute(DCTERMS.identifier)
+        final_dataset = await self.get(dataset_id, context)
         return final_dataset, errors
 
-    async def get(self, id: str, context: Context) -> Dataset:
-        """Get a dataset by its ID"""
-        query = FilterDatasetByID(id)
-        return await self.repositories.datasets.get(query)
-
-    async def delete(self, id: str, context: Context) -> None:
-        """Delete a dataset by its ID"""
-        query = FilterDatasetByID(id)
-        await self.repositories.datasets.delete(query)
+    def _parse_access_url(self, file_path: str) -> tuple[str, str]:
+        """
+        Parses accessURL into (interface, encoded_resource_path).
+        Examples:
+            file://disease_xyz/images/patient1.dcm →
+             ("file", "disease_xyz/images/Fpatient1.dcm")
+            file:///data/disease_xyz/images/patient1.dcm →
+             ("file", "disease_xyz/Fimages/patient1.dcm")
+            s3://bucket/key/file.csv → ("s3", "bucket/Fkey/Ffile.csv")
+            https://example.com/data/file.csv  →
+            ("http", "https://example.com/data/file.csv")
+        """
+
+        parsed = urlparse(file_path)
+
+        if parsed.scheme == "file":
+            path_part = file_path.split("://", 1)[-1]
+
+            # If it starts with /data/, trim only that prefix
+            if path_part.startswith("/data/"):
```

Comment:
This shouldn't be static like this for now we are using data folder but it could change, it should depend on constant that is defined and it can be read from your env. variables